### ISSUE TITLE
Review naming convention for named pipes

### ISSUE THREAD

**Issue by lmorg** [lmorg/murex#104]
Currently only alpha and numeric characters are allowed (which is super non-obvious when using the `pipe` function as that allows any input without error) but it might be nice to support full stop / period character for name spaces - a bit like the naming convention for `private` functions.

**Comment by lmorg** [comment]
I think this definitely needs to happen as part of a larger piece of work where I also strip pipes from parameters if the parameters are not quoted (so `out "<html>"`, for example, doesn't try to pipe to `html`). This has been a long standing annoyance of mine!

Also it might be nice to support temporary named pipes. eg:
`pretty: <file:~/something.json>` works instead of `<stdin> -> pretty -> > ~/something.json`

And support for STDIN too? eg `<|pipe>` or `<=pipe>`? My preference would be `<|pipe>` but I think that might cause havoc with the parsers and syntax highlighters so `<=pipe>` would be easier to implement.

**Comment by lmorg** [comment]
The preliminary work has been committed as part of https://github.com/lmorg/murex/pull/145

**Comment by lmorg** [comment]
Removing the `bug` label because this is now a feature request

**Comment by lmorg** [comment]
Temporary named pipes are now supported (albeit name might change).

A couple of examples
```
out: <tcp-dial:murex.rocks:80> GET /

cat: <file:copy.md> README.md
```

---

I don't particularly fancy using the same format to pull from STDIN. I think it would result in less readable code for no advantage. However if I were to I'd probably use `<-pipe>`

**Comment by lmorg** [comment]
Additional support has now been included in v2.2 to set environmental variables per executable.

Bash
```
$ foo=bar env | grep foo
foo=bar
```

Murex
```
$ env <env:foo=bar> | grep foo
foo=bar
```
